chat-widget/src/components/livechatwidget/livechatwidgetstateful/LiveChatWidgetStateful.tsx

```diff
@@ -243,6 +246,21 @@ export const LiveChatWidgetStateful = (props: ILiveChatWidgetProps) => {
                 if (props.controlProps?.hideStartChatButton) {
                     dispatch({ type: LiveChatWidgetActionType.SET_MINIMIZED, payload: event?.payload?.isChatHidden });
                 }
+                const dateNow = Date.now();
+                if (dateNow - lastLWICheckTimeRef.current > Constants.LWICheckOnVisibilityTimeout) {
+                    chatSDK?.getConversationDetails()
+                        .then(async(conversationDetails: LiveWorkItemDetails) => {
+                            lastLWICheckTimeRef.current = dateNow;
+                            if (conversationDetails &&
+                                (conversationDetails?.state === LiveWorkItemState.WrapUp || conversationDetails.state === LiveWorkItemState.Closed)) {
+                                dispatch({ type: LiveChatWidgetActionType.SET_CHAT_DISCONNECT_EVENT_RECEIVED, payload: true });
+                                TelemetryHelper.logActionEvent(LogLevel.INFO, {
+                                    Event: TelemetryEvent.chatDisconnectThreadEventReceived,
+                                    Description: "Chat disconnected due to timeout, left or removed."
+                                });
+                            }
+                        });
+                }
```

Comment:
Is this the right place to do, please check the performance impact.

Reply:
Yes, this the place where we are subscribing for visibility change event. Also, the call to get conversation details is made after every 3 minutes